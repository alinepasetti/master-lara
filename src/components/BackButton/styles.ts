import styled, { css } from 'styled-components';
import { BiArrowBack } from 'react-icons/bi';

export const Back = styled(BiArrowBack)`${({ theme }) => css`
  width: 40px;
  height: 40px;
  color: ${theme.colors.mediumGray};
  background-color: ${theme.colors.white};
  font-size: 0.8rem;
  border-radius: 50%;
  padding: 0.7rem;
  position: absolute;
  left: 1.6rem;
  top: 1.6rem;
`}`;
