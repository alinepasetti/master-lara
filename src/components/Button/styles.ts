import styled, { css } from 'styled-components';

export const Button = styled.a` ${({ theme }) => css`
  width: 90%;
  display: block;
  padding: 1.6rem;
  font-size: 2.4rem;
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
  border: none;
  margin: 0 auto 2.4rem;
  text-align: center;
  text-decoration: none;
`}`;
