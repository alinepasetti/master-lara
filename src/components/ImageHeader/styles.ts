import styled from 'styled-components';

export const Image = styled.img`
  width: 100%;
  height: 40vh;
  object-fit: cover;
  object-position: center;
`;
