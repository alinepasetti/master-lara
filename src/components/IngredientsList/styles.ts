import styled, { css } from 'styled-components';
import { AiFillCloseCircle } from 'react-icons/ai';
import { AiFillCheckCircle } from 'react-icons/ai';

type HasIngredient = {
  hasIngredient: boolean;
};

const uncheckIconUrl = (theme) =>
  `url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 24 24' stroke-width='1.5' stroke='%23${theme.colors.secondaryColor.substring(
    1,
  )}' class='w-6 h-6'%3E%3Cpath stroke-linecap='round' stroke-linejoin='round' d='M9.75 9.75l4.5 4.5m0-4.5l-4.5 4.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z' /%3E%3C/svg%3E")`;

const checkIconUrl = (theme) =>
  `url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 24 24' stroke-width='1.5' stroke='%23${theme.colors.green.substring(
    1,
  )}' class='w-6 h-6'%3E%3Cpath stroke-linecap='round' stroke-linejoin='round' d='M9 12.75L11.25 15 15 9.75M21 12a9 9 0 11-18 0 9 9 0 0118 0z' /%3E%3C/svg%3E%0A")`;

export const Container = styled.ul`
  padding: 2.4rem;
  padding-top: 0;
  margin: 0;

  &> a {
    margin-top: 1.6rem;
  }
`;

export const ListItem = styled.li<HasIngredient>`${({
  theme,
  hasIngredient,
}) => css`
  list-style: none;

  &::before {
    display: inline-block;
    content: ${hasIngredient ? checkIconUrl(theme) : uncheckIconUrl(theme)};
    width: 1.8rem;
    padding-right: 1rem;
}
`}`;

export const CheckIcon = styled(AiFillCheckCircle)`${({ theme }) => css`
  color: ${theme.colors.green};
  margin-right: 2.4rem;
  height: 2.4rem;
  width: 2.4rem;
`}`;

export const UncheckIcon = styled(AiFillCloseCircle)`${({ theme }) => css`
  color: ${theme.colors.secondaryColor};
  margin-right: 2.4rem;
  height: 2.4rem;
  width: 2.4rem;
`}`;
