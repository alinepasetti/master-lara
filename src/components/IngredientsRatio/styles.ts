import styled, { css } from 'styled-components';

export const Container = styled.p`${({ theme }) => css`
  font-size: 1.6rem;
  margin: 0 0;
  color: ${theme.colors.white};
  background-color: ${theme.colors.primaryColor};
  font-size: 1.2rem;
  font-weight: 400;
  position: absolute;
  right: 1.6rem;
  top: 1.6rem;
  border-radius: 8px;
  box-shadow: 0 0 5px ${theme.colors.mediumGray};
  padding: 1rem;
`}`;
