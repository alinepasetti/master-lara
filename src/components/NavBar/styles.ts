import styled from 'styled-components';

export const NavBar = styled.nav`
  width: 100%;
  padding: 1rem 1.6rem;
  background-color: black;
`;

export const Image = styled.img`
  width: 20%;
  display: block;
  margin: 0 auto;
`;
