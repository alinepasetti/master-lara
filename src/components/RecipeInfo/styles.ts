import styled from 'styled-components';
import { Title as Heading } from '../Heading/styles';

export const Container = styled.div`

  padding: 1.6rem

  ${Heading} {
    font-size: 1.6rem;
    margin: 0 0;
  }
`;

export const Tag = styled.h3`
  font-size: 1.6rem;
  margin: 0 0;
  color: ${({ theme }) => theme.colors.mediumGray};
  font-weight: 400;
`;
