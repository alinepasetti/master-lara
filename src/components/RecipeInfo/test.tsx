import { screen } from '@testing-library/react';
import { RecipeInfo } from '.';
import { renderTheme } from '../../styles/render-theme';

describe('<RecipeInfo />', () => {
  it('should render a heading and tags', () => {
    renderTheme(
      <RecipeInfo
        label="couscous"
        cuisineType="Moroccan"
        totalTime={60}
        ingredientRatio="1 / 3"
      />,
    );
    expect(
      screen.getByRole('heading', { name: 'couscous' }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { name: 'Moroccan' }),
    ).toBeInTheDocument();
    expect(screen.getByRole('heading', { name: '1 hour' })).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { name: '1 / 3 ingredients' }),
    ).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    renderTheme(
      <RecipeInfo
        label="couscous"
        cuisineType="Moroccan"
        totalTime={60}
        ingredientRatio="1 / 3"
      />,
    );
    expect(
      screen.getByRole('heading', { name: 'couscous' }).parentElement,
    ).toMatchSnapshot();
  });
});
