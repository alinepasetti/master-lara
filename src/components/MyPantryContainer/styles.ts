import styled from 'styled-components';
import { Button as SearchButton } from '../Button/styles';

export const Container = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;

  ${SearchButton} {
    border-radius: 8px;
    width: 10.8rem;
  }
`;
