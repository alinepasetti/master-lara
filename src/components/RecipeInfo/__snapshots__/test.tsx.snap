// Vitest Snapshot v1

exports[`<RecipeInfo /> > should match snapshot 1`] = `
.c0 {
  padding: 2.4rem;
  position: relative;
}

.c1 {
  font-size: 1.6rem;
  margin: 0 0;
  color: #5c5c5c;
  font-weight: 400;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1> svg {
  margin-right: 0.5rem;
}

.c2 {
  border: 0;
  border-bottom: 1px solid #5c5c5c;
  margin: 2.4rem auto 0 auto;
  width: 98%;
}

<div
  class="c0"
>
  <h1
    class=""
  >
    couscous
  </h1>
  <h3
    class="c1"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M39.906 27.188c-9.118 13.907-11.366 30.99-7.843 50.718 4.2 23.526 16.91 50.038 35.28 75 36.742 49.925 96.05 93.082 148.813 99.625l3.688.47 2.375 2.844L416.374 490.22c19.352 4.624 31.847 1.745 38.344-4.69 6.547-6.484 9.566-19.005 4.717-38.874L39.908 27.187zM414.97 29.5L306.47 138c-12 11.998-12.104 25.2-5.908 39.625l2.563 5.97-4.688 4.5L262 222.844l29.594 29.593 34.594-36.532 4.5-4.75 5.968 2.594c15.165 6.535 29.546 6.267 40.688-4.875l108.5-108.5L471.75 86.28l-70.563 70.532L388 143.595l70.53-70.53L443.5 58.03l-70.53 70.532-13.22-13.218 70.53-70.53-15.31-15.314zM210.936 271.563L25.53 448.469c-4.575 18.95-1.644 30.787 4.532 36.905 6.178 6.118 18.128 8.927 36.844 4.406l173.22-182.967-29.19-35.25z"
      />
    </svg>
    Moroccan
  </h3>
  <h3
    class="c1"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm176.5 585.7l-28.6 39a7.99 7.99 0 0 1-11.2 1.7L483.3 569.8a7.92 7.92 0 0 1-3.3-6.5V288c0-4.4 3.6-8 8-8h48.1c4.4 0 8 3.6 8 8v247.5l142.6 103.1c3.6 2.5 4.4 7.5 1.8 11.1z"
      />
    </svg>
    1 hour
  </h3>
  <hr
    class="c2"
  />
</div>
`;
